File: Java/Container-With-Most-Water.java
public class Solution {
    /**
     * @param heights: an array of integers
     * @return: an integer
     */
    public int maxArea(int[] heights) {
        // write your code here
        int maxV = 0, left = 0, right = heights.length - 1;
        while (left < right) {
            maxV = Math.max(maxV, (right - left) * (Math.min(heights[left], heights[right])));
            if (heights[left] <= heights[right]) left ++;
            else right --;
        }
        return maxV;
    }
}

File: Python/Container-With-Most-Water.py
class Solution:
    # @param heights: a list of integers
    # @return: an integer
    def maxArea(self, heights):
        # write your code here
        maxarea = 0
    	left = 0
        right = len(heights) - 1
        while left < right:
            maxarea = max(maxarea, min(heights[left], heights[right]) * (right - left))
            if heights[left] < heights[right]:
                left += 1
            else:
                right -= 1
        return maxarea